### ISSUE TITLE
release lock on SIGTERM too.

### ISSUE THREAD

**Issue by mildred** [gruntwork-io/terragrunt#61]
On SIGINT (Interrupt from keyboard), terragrunt release the lock, but not on SIGTERM (Termination signal). This should probably be fixed. See `locks/lock.go`.

```
diff --git a/locks/lock.go b/locks/lock.go
index 58ace31..13af32b 100644
--- a/locks/lock.go
+++ b/locks/lock.go
@@ -5,6 +5,7 @@ import (
        "github.com/gruntwork-io/terragrunt/errors"
        "os"
        "os/signal"
+       "syscall"
 )
 
 // Every type of lock must implement this interface
@@ -48,6 +49,7 @@ func WithLock(lock Lock, action func() error) (finalErr error) {
        // the blocking call to action() to return normally.
        signalChannel := make(chan os.Signal, 1)
        signal.Notify(signalChannel, os.Interrupt)
+       signal.Notify(signalChannel, syscall.SIGTERM)
        go func() { util.Logger.Printf("Caught signal '%s'. Terraform should be shutting down gracefully now.", <- signalChannel) }()
 
        return action()
```

**Comment by mildred** [comment]
Additionally, terragrunt should forward SIGTERM signals to terraform. There is no guaantee that terraform will have received it. See also hashicorp/terraform#10459

**Comment by mildred** [comment]
In fact, on some cases (default on systemd services) SIGTERM is sent to all processes in the cgroup. In other cases (sysvservice, docker) the SIGTERM signal is only sent to the parent process.

**Comment by brikis98** [comment]
Good point. Looks like you have some code written for this already. Would you like to submit a PR?

**Comment by mildred** [comment]
Yes, I will submit a PR once I get something with terraform. There are two aspects:

- handle SIGTERM as well as SIGINT
- make sure terraform has received SIGTERM. When we type Ctrl-C in a terminal, the VT will make sure that all VT clients are sent a SIGINT signal. With SIGTERM, we are not sure that terraform received SIGTERM and we must probably forward the signal.

Are you ok with forwarding the SIGTERM signal ?

When forwarding SIGTERM we must make sure terraform can handle it. Currently it does not respond to SIGTERM and doesn't cleanly finish like it does with Ctrl-C. I'm currently submitting a PR on terraform for that.

**Comment by brikis98** [comment]
Probably makes sense to wait for your PR to be merged into Terraform itself.

**Comment by mildred** [comment]
yes

**Comment by brikis98** [comment]
This should have been fixed by #72.
